### PR TITLE
slcli bandwidth pools-create with single locations as option #1990

### PR DESCRIPTION
**Pull request by JayasilanJibm**
slcli bandwidth pools-create with single locations as option

Fixes: #1990

This PR provides users with an option to specify single as well as multiple regions.

For example. 

SJC/DAL/WDC/TOR/MON A user can specify any of those to get a pool in that region.
slcli bandwidth pools-create --region MON
slcli bandwidth pools-create --region SJC/DAL/WDC/TOR/MON

Both of the above commands will work.

Thanks,
Jayasilan

### CODE REVIEW

**Comment by allmightyspiff** [comment]
image
Dont manually add the console color codes.

File: SoftLayer/CLI/bandwidth/pools_create.py [hunk 1]
```diff
@@ -21,13 +21,36 @@
...
+
 
 @click.command(cls=SLCommand)
 @click.option('--name', required=True, help="Pool name")
 @click.option('--region', required=True,
-              type=click.Choice(['SJC/DAL/WDC/TOR/MON', 'AMS/LON/MAD/PAR', 'SNG/HKG/OSA/TOK',
-                                'SYD', 'MEX', 'SAO', 'CHE', 'MIL', 'SEO', 'FRA']),
-              help="Region selected")
+              help=f"Choose Region/Region-Group {regions}", callback=check_region_param)
+@click.help_option('--help', '-h', help=f"Specify Region or Region group - \033[1;32m{HELP_STMT}")
```
Comment (allmightyspiff) [inline SoftLayer/CLI/bandwidth/pools_create.py:53]: Dont overwrite the default help text. Add it instead to the Doc Block comment, see comment below.

File: SoftLayer/CLI/bandwidth/pools_create.py [hunk 2]
```diff
@@ -21,13 +21,36 @@
...
 @environment.pass_env
 def cli(env, name, region):
     """Create bandwidth pool."""
```
Comment (allmightyspiff) [inline SoftLayer/CLI/bandwidth/pools_create.py:56]:
```suggestion
    """Create bandwidth pool.
    
    Region can be the full zone name 'SJC/DAL/WDC/TOR/MON', or just a single datacenter like 'SJC'.

    Example::

        slcli bandwidth pool-create --name testPool --region DAL
        slcli bandwidth pool-create --name testPool --region SJC/DAL/WDC/TOR/MON
```

`Example::` is a format that will get picked up by the documentation generator to make it a nice formatted table, like https://softlayer-python.readthedocs.io/en/latest/cli/vs/#virtual-bandwidth

File: SoftLayer/CLI/bandwidth/pools_create.py [hunk 3]
```diff
@@ -21,13 +21,36 @@
     "FRA": "FRA"
 }
 
+regions = ['SJC/DAL/WDC/TOR/MON', 'AMS/LON/MAD/PAR', 'SNG/HKG/OSA/TOK', 'SYD', 'MEX', 'SAO', 'CHE', 'MIL', 'SEO', 'FRA']
+HELP_STMT = "Ex: 'HKG' or 'SNG/HKG/OSA/TOK'"
+
+
+def check_region_param(ctx, param, value):
```
Comment (allmightyspiff) [inline SoftLayer/CLI/bandwidth/pools_create.py:28]:
```suggestion
def check_region_param(ctx, param, value):  # pylint: disable=unused-argument
```
this disables the error about unused params, so you can remove the `_ = [ctx, param]` line.

File: SoftLayer/CLI/bandwidth/pools_create.py [hunk 4]
```diff
@@ -21,13 +21,36 @@
...
+        if value in key or value is key:
+            region_group = key
+        else:
+            continue
+
+    if region_group:
+        return region_group
+    else:
+        raise click.BadParameter(f"{value} is not a region or part of any region.\
+                                 \nAvailable Choices: \033[1;32m{regions}")
```
Comment (allmightyspiff) [inline SoftLayer/CLI/bandwidth/pools_create.py:46]:
```suggestion
        raise click.BadParameter(f"{value} is not a region or part of any region."
                                 " Available Choices: {regions}")
```

File: SoftLayer/CLI/bandwidth/pools_create.py [hunk 5]
```diff
@@ -21,13 +21,36 @@
...
+    # :params string value: Region or Region-Groups
+    # return string Region-Groups
+
+    _ = [ctx, param]
+    region_group = None
+    for key in location_groups:
+        if value in key or value is key:
+            region_group = key
+        else:
+            continue
```
Comment (allmightyspiff) [inline SoftLayer/CLI/bandwidth/pools_create.py:40]: Is this line needed? If not I think we can remove it.

**Review by allmightyspiff — CHANGES_REQUESTED** [review]
Looks good, see comments. Also I'd like to add a unit test specifically for the case where a user selects a region like `SJC` and make sure the CLI sends the correct region to the manager.

File: SoftLayer/CLI/bandwidth/pools_create.py (hunk 5 above)
Comment (JayasilanJibm) [reply SoftLayer/CLI/bandwidth/pools_create.py]: Removed and tested

**Comment by ramkishor-ch** [comment]
Hi,

I have tested the changes.
Please find the screen shot for reference.

Screen Shot:
Screenshot 2023-06-27 at 5 09 21 PM

Thanks,
Ramkishor Chaladi.

**Review by ramkishor-ch — APPROVED** [review]

**Review by allmightyspiff — APPROVED** [review]

**Pull request merged**